komodo/symlink/create_links.py

```diff
@@ -0,0 +1,61 @@
+import os,sys
+import argparse
+import json
+from .sanity_check import verify_integrity
+from contextlib import contextmanager
+
+@contextmanager
+def working_dir(path):
+    prev_dir = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(prev_dir)
+
+def _create_link(src, dst, link_dict):  
+    if src in link_dict and not os.path.exists(src):
+        _create_link(link_dict[src], src, link_dict)
+
+    if not os.path.exists(src):
+        raise ValueError("{} does not exist".format(src))
+    
```

Comment:
To minimise the number of touches on the disk and the chances for syncing something broken, should we do something like:

```python
if os.path.exists(dst) and os.path.islink(dst):
    if os.readlink(dst) == src:
        return
    else:
        os.remove(dst)
```

Reply:
:+1: